ui/app/routes/vault/cluster/clients/index.js

```diff
@@ -43,20 +43,44 @@ export default Route.extend(ClusterRoute, {
     },
   },
 
-  model(params) {
+  async getLicense() {
+    try {
+      return await this.store.queryRecord('license', {});
+    } catch (e) {
+      // ARG TODO handle
+      return e;
+    }
+  },
+
+  async getNewInitActivity(start_time) {
+    try {
+      return await this.store.queryRecord('clients/newInitActivity', { start_time });
+    } catch (e) {
+      // ARG TODO handle
+      return e;
+    }
+  },
+
+  // ARG POC
+  async model(params) {
     let config = this.store.queryRecord('clients/config', {}).catch((e) => {
       console.debug(e);
       // swallowing error so activity can show if no config permissions
       return {};
     });
     const activityParams = getActivityParams(params);
-    let activity = this.store.queryRecord('clients/activity', activityParams);
+    let activity = this.store.queryRecord('clients/activity', activityParams); // ARG TODO should async await and try catch
+    let license = await this.getLicense();
+    let newInitActivity = await this.getNewInitActivity(license.startTime);
```

Comment:
What are your thoughts on opting to use _either_ the `hash` method to resolve all of the promises or async/await? I think picking one or the other would make things more clear. The `hash` method seems fine if you don't need to handle any of the errors individually. In that case if you were to move the logic to fetch the license inside the `getNewInitActivity` method then you could remove the awaits.

Reply:
Totally agree. I have the comment to remove hash once we remove activity. The problem now is it borks things if I try and take it out and/or use activity as a async/await.